src/magic.cpp

```diff
@@ -1415,6 +1475,28 @@ bool known_magic::has_enough_energy( const player &p, spell &sp ) const
             return false;
         case fatigue_energy:
             return p.get_fatigue() < EXHAUSTED;
+        case item_energy: {
+            const itype_id id = sp.energy_item();
+            std::vector<const item*> a_filter = p.inv.items_with( [id](const item& it ) {
```

Comment:
You do not need to reinvent the wheel: use `inventory::has_components` or `inventory::has_charges` to check the amount of available items (see `item_comp::has` in "src/requirements.cpp" for an example).

Reply:
I was wondering where those functions were, and what they were called, thanks.